kornia/color/zca.py

```diff
@@ -156,30 +162,37 @@ def zca_mean(inp: torch.Tensor, dim: int = 0,
              return_inverse: bool = False) -> Tuple[torch.Tensor, torch.Tensor, Optional[torch.Tensor]]:
     r"""
 
-    Computes ZCA whitening matrix and mean vector. The output could be used in
-    :class:`~torchvision.transforms.LinearTransformation` or with
-    :method:`~kornia.color.zca_whiten`.
+    Computes ZCA whitening matrix and mean vector. The output can be used with
+    :py:meth:`~kornia.color.linear_transform`
 
     See :class:`~kornia.color.ZCAWhitening` for details.
 
 
     args:
         inp (torch.Tensor) : input data tensor
         dim (int): Specifies dimension that serves as samples dimension. Default = 0
-        eps (float) : a small number used for numerial stablility. Default = 0
+        eps (float) : a small number used for numerical stablility. Default = 0
         unbiased (bool): Whether to use the biased estimate of the covariance matrix. Default = True
-        return_inv (bool): Whether to return the inverse ZCA transform.
-
+        return_inverse (bool): Whether to return the inverse ZCA transform.
 
+    shapes:
+        - inp: :math:`(D_0,...,D_{\text{dim}},...,D_N)` is a batch of N-D tensors.
+        - transform_matrix: :math:`(\Pi_{d=0,d\neq \text{dim}}^N D_d, \Pi_{d=0,d\neq \text{dim}}^N D_d)`
+        - mean_vector: :math:`(1, \Pi_{d=0,d\neq \text{dim}}^N D_d)`
+        - inv_transform: same shape as the transfrom matrix
 
     returns:
         Tuple[torch.Tensor, torch.Tensor, torch.Tensor]:
         A tuple containing the ZCA matrix and the mean vector. If return_inv is set to True,
         then it returns the inverse ZCA matrix, otherwise it returns None.
 
     Examples:
+        >>> from kornia.color import zca_mean
         >>> x = torch.tensor([[0,1],[1,0],[-1,0],[0,-1]], dtype = torch.float32)
-        >>> transform, mean = zca_mean(x) # Returns transformation matrix and data mean
+        >>> transform_matrix, mean_vector,_ = zca_mean(x) # Returns transformation matrix and data mean
+        >>> x = torch.rand(3,20,2,2)
+        >>> transform_matrix, mean_vector, inv_transform = zca_mean(x, dim = 1, return_inverse = True)
+        >>> # transform_matrix.size() equals (12,12) and the mean vector.size equal (1,12)
```

Comment:
just curious, are you actually testing this in some way ? we are planning to include to doctest to keep documentation prune from errors

Reply:
I copy paste them into the ipython repl to verify if they work.

Reply:
it's fine by now, we are figuring out how to properly test documentation since it can be consistant with potential changes

Reply:
@edgarriba Just a heads up. I made some grammar and spelling fixes, and I tried to improve the clarity of the documentation for zca whitening in PR #567. Sorry I missed these I was pretty sleepy last night, and I wanted to put something out there given the existing documentation wasn't correct in some spots in the initial PR.